components/psst/browser/content/psst_tab_helper.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_TAB_HELPER_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_TAB_HELPER_H_
+
+#include "base/component_export.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+
+#include "brave/components/psst/browser/core/psst_rule.h"
+#include "build/build_config.h"
+#include "components/sessions/core/session_id.h"
+#include "content/public/browser/media_player_id.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace psst {
+
+class PsstRuleService;
+
+class COMPONENT_EXPORT(PSST_BROWSER_CONTENT) PsstTabHelper
+    : public content::WebContentsObserver,
+      public content::WebContentsUserData<PsstTabHelper> {
+ public:
+  static void MaybeCreateForWebContents(content::WebContents* contents,
+                                        const int32_t world_id);
+  ~PsstTabHelper() override;
+  PsstTabHelper(const PsstTabHelper&) = delete;
+  PsstTabHelper& operator=(const PsstTabHelper&) = delete;
+
+ private:
+  COMPONENT_EXPORT(PSST_BROWSER_CONTENT)
```

Comment:
this doesn't seem like it needs to be here? Were you getting an error without it?